index.html

```diff
@@ -557,82 +442,106 @@ <h3>
+            </li>
+            <li data-tests="wakelock-disabled-by-feature-policy.https.sub.html">
+            If |document| is not <a>allowed to use</a> the <a>policy-controlled
+            feature</a> named "`wake-lock`", reject |promise| with a
+            "{{NotAllowedError}}" {{DOMException}} and return |promise|.
+            </li>
+            <li>If the <a>user agent</a> <a data-lt="deny wake lock">denies the
+            wake lock</a> of this |type| for |document|, reject |promise|
+            with a "{{NotAllowedError}}" {{DOMException}} and return |promise|.
+            </li>
+          </ol>
+          <li>If the <a>current global object</a> is the
+          {{DedicatedWorkerGlobalScope}} object:
             <ol>
-              <li>Run <a>release a wake lock</a> on |lock|.
+              <li>If the <a>current global object</a>'s <a>owner set</a>
+              <a data-xref-for="list">is empty</a>, reject |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and return |promise|.
+              </li>
+              <li data-link-for="">If |type| is {{
+              WakeLockType["screen"] }}, reject
+              |promise| with a "{{NotAllowedError}}" {{DOMException}},
+              and return |promise|.
               </li>
             </ol>
           </li>
-          <li>Set |lock|.{{[[requestPromise]]}} to promise.
+          <li>Otherwise, if the <a>current global object</a> is the {{Window}} object:
+            <ol>
+              <li>If the |document|'s [=Document/browsing context=] is `null`, reject
+              |promise| with a "{{NotAllowedError}}" {{DOMException}} and
+              return |promise|.
+              </li>
+              <li>If |document| is not <a>fully active</a>, reject
+              |promise| with a "{{NotAllowedError}}" {{DOMException}},
+              and return |promise|.
+              </li>
+              <li data-link-for="">If |type| is {{
+              WakeLockType["screen"] }} and the {{Document}} of the
+              <a>top-level browsing context</a> is <a>hidden</a>, reject
+              |promise| with a "{{NotAllowedError}}" {{DOMException}},
+              and return |promise|.
+              </li>
+            </ol>
           </li>
           <li>Return |promise| and run the following steps <a>in parallel</a>:
             <ol>
-              <li>Let |state:PermissionState| be the result of running and waiting for the <a>
-              obtain permission</a> steps with |lock|.|type:WakeLockType|.
+              <li data-link-for="WakeLockRequestOptions">
+              Let |signal:AbortSignal| be |options|'s {{signal}} member if
+              present, or `null`.
               </li>
-              <li>If |state| is {{"denied"}}, then reject |promise| with a
-              "{{NotAllowedError}}" {{DOMException}}, and run <a>abort the
-              request steps</a>.
+              <li>If |signal| is not `null` and |signal|’s <a>aborted flag</a>
```

Comment:
This check is being repeated 3 times in the parallel steps. Moreover, now that the list of steps has changed quite a bit, I think it makes sense to split the `signal` checks:
* If `signal` is set and its aborted flag is set, reject and abort the steps earlier so that we don't even enter the "parallel steps" part.
* If it's just set but the abort flag is not set, then add "release wake lock" to `signal`s list of algorithms as the last parallel step (or at least after running "acquire wake lock").

Reply:
> * If it's just set but the abort flag is not set, then add "release wake lock" to `signal`s list of algorithms as the last parallel step (or at least after running "acquire wake lock").

Actually, since the "release wake lock" algorithm aborts early if `promiseLock` is not present in `[[ActiveLocks]]`, it's also possible to just move all the `signal` handling out of the parallel steps.

Reply:
I am not really following you here, but maybe the best would be landing this and they you can create a follow up patch?

Reply:
Done in #209